zerver/lib/email_notifications.py

```diff
@@ -247,6 +247,13 @@ def build_message_payload(message: Message, sender: Optional[str] = None) -> Dic
         relative_to_full_url(fragment, user.realm.uri)
         fix_emojis(fragment, user.emojiset)
         fix_spoilers_in_html(fragment, user.default_language)
+
+        # Following changes are done to avoid the bad rendering of latex math formulas in emails.
+        # See: #25289 for more details.
```

Comment:
I think a much better comment here would be one that explains the structure of KaTeX output and thus how we're going the algorithm below will get the plain-text version of the KaTeX input.